pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -265,20 +279,83 @@ func (e *etcd) Deploy(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, e.client, networkPolicy, func() error {
-		networkPolicy.Annotations = map[string]string{
+	// add peer certs if shoot has HA control plane
+	var (
+		etcdPeerCASecret *corev1.Secret
+		peerServerSecret *corev1.Secret
+	)
+	if _, ok := e.annotations[v1beta1constants.ShootAlphaControlPlaneHighAvailability]; ok && gardenletfeatures.FeatureGate.Enabled(features.HAControlPlanes) {
```

Comment:
> I do acknowledge that changing this to smaller functions would be more effort. So you can take a call if this adds value.

I would suggest to do this outside of this PR since it's not strictly required now, and personally I don't see a problem with it right now. However, I like the suggestion with initializing some bool var or field in `etcd` to not replicate the check again and again.

Reply:
Yes, I agree. I'll introduce `NodeSpread` and `ZoneSpread` fields in the `etcd` struct, and reuse these in the code. Thanks.